### PR TITLE
Adyen - dont put discount amount if its empty

### PR DESCRIPTION
**Pull request by nirajjayant**

### CODE REVIEW

File: gateways/adyen/request_builder.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ func buildLevel3Data(level3Data *sleet.Level3Data) map[string]string {
 			break
 		}
 		keyBase = fmt.Sprintf("enhancedSchemeData.itemDetailLine%d.", idx+1)
+		// Due to issues with the credit card networks, dont send any line item if discount amount is 0
+		if lineItem.ItemDiscountAmount.Amount > 0 {
+			additionalData[keyBase+"discountAmount"] = sleet.AmountToString(&lineItem.ItemDiscountAmount)
```
Comment (namanhkapur) [inline gateways/adyen/request_builder.go:108]: nit should we add whitespace around `+`?

File: gateways/adyen/request_builder_test.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ func TestBuildAuthRequest(t *testing.T) {
 	base := sleet_testing.BaseAuthorizationRequest()
 	requestWithLevel3Data := sleet_testing.BaseAuthorizationRequest()
 	requestWithLevel3Data.Level3Data = sleet_testing.BaseLevel3Data()
+	requestWithLevel3ItemDiscount := sleet_testing.BaseAuthorizationRequest()
+	requestWithLevel3ItemDiscount.Level3Data = sleet_testing.BaseLevel3Data()
+	requestWithLevel3ItemDiscount.Level3Data.LineItems[0].ItemDiscountAmount.Amount = 100
```
Comment (namanhkapur) [inline gateways/adyen/request_builder_test.go:21]: where did the `100` come from?

**Review by namanhkapur — APPROVED** [review]
LGTM

File: gateways/adyen/request_builder_test.go (hunk 2 above)
Comment (nirajjayant) [reply gateways/adyen/request_builder_test.go:21]: Just some random number haha

File: gateways/adyen/request_builder.go (hunk 1 above)
Comment (nirajjayant) [reply gateways/adyen/request_builder.go:108]: Like idx + 1? Sure

File: gateways/adyen/request_builder_test.go (hunk 2 above)
Comment (namanhkapur) [reply gateways/adyen/request_builder_test.go:21]: oh i didn't realize this was a test whoops

**Pull request merged**